### PR TITLE
Quiet option to swallow misconfiguration

### PR DESCRIPTION
**Pull request by tpitale**
Related to https://github.com/tpitale/mail_room/pull/24

/cc @DouweM.

### CODE REVIEW

**Comment by tpitale** [comment]
@DouweM if this sufficiently handles the situation you highlighted here (https://github.com/tpitale/mail_room/pull/24#discussion_r37367540), just give me a thumbs up and I'll merge this.

**Comment by tpitale** [comment]
Also resolves https://github.com/tpitale/mail_room/issues/25

File: lib/mail_room/configuration.rb [hunk 1]
```diff
@@ -2,16 +2,21 @@ module MailRoom
...
+      self.quiet = options.fetch(:quiet, false)
 
       if options.has_key?(:config_path)
-        config_file = YAML.load_file(options[:config_path])
+        begin
+          config_file = YAML.load_file(options[:config_path])
 
-        set_mailboxes(config_file[:mailboxes])
+          set_mailboxes(config_file[:mailboxes])
+        rescue => e
```
Comment (DouweM) [inline lib/mail_room/configuration.rb]: Is it smart to rescue _any_ error? I think we just want to quiet non existing/empty config files, not anything at all that goes wrong inside mail_room.

Comment (tpitale) [inline lib/mail_room/configuration.rb]: This only rescues from the configuration portion, nothing to do with _running_ MailRoom.

Comment (DouweM) [inline lib/mail_room/configuration.rb]: Ah, that's right. Still, something like this seems cleaner:

```ruby
return if quiet && !File.exist?(options[:config_path]
return if quiet && config_file[:mailboxes].nil?
```

Comment (tpitale) [inline lib/mail_room/configuration.rb]: That only handles the two specific cases you're worried about. I'd like to handle other cases in configuration. To explicitly check with guard statements for every one of those cases does not seem cleaner to me. Also, not a fan of guard statements, especially stacked ones.

**Comment by tpitale** [comment]
Oh, related question: do you want mail_room to run if there are no mailboxes? Or do you think it should just return from the `coordinator` immediately?

File: lib/mail_room/configuration.rb (hunk 1 above)
Comment (DouweM) [inline lib/mail_room/configuration.rb]: @tpitale Fair enough.

**Comment by DouweM** [comment]
@tpitale I think quitting immediately makes most sense. But we'll probably use some god process to make sure our `mail_room` process is automatically restarted when it crashes, and I'll have to see how that behaves when `mail_room` can also quit immediately and doesn't need to be restarted.

**Comment by tpitale** [comment]
Yeah, that's my concern about `-q` plus quitting immediately if no mailboxes are registered from configuration. It might just be an eternal loop of restarting. Not sure that silently running in a loop sleeping for 1 second at a time is much better.

I should probably add something about the dangers of `-q` and misconfiguration when used with upstart/init/god/monit etc.

**Comment by DouweM** [comment]
> Not sure that silently running in a loop sleeping for 1 second at a time is much better.

While non-obvious, I think it'll be less expensive to the CPU.

**Comment by tpitale** [comment]
To let a process manager handle restarting? Or letting ruby sleep 1 in a loop?

I was going to make a note about configuring upstart et al to use incremental backoff when restarting processes: https://github.com/tpitale/mail_room/issues/31. Like, upstart can be configured to only try to restart 5x in a minute or something.

**Comment by DouweM** [comment]
@tpitale Sorry, I mean it's probably less expensive to let Ruby sleep 1 second in a loop. The process manager will have more overhead, and it won't activate at most once per second. Incremental back off/throttling sounds like a good solution. 

Something I don't know is if these process managers care about the process exit code. It'd be nice if they didn't try to restart if `mail_room` exited with `0`.

**Comment by tpitale** [comment]
That's a good question, re exit status. I'll try to check for upstart. If there's a special code, we can exit with that.

**Pull request merged**